Added functionality for the edgeos_config module to list and delete unmanaged config

graemedavidson opened the pull request:
##### SUMMARY
Added the ability for the edgeos_config module to return a list of unmanaged config. A flag can also be passed to allow this unmanaged config to be changed into delete statements. 

By adding this functionality it provides a method of moving towards a one to one relation between code and active config. 

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
edgeos_config

##### ADDITIONAL INFORMATION
- Have also included some general updates to variable names and test formatting to improve readability 
- Included a warning and list for invalid candidate commands

Commit:
updated test format and variable naming for better readability to
edgeos_config module

Added filtered tests

Commit:
added docstring to edgeos_config module

Commit:
added ability to list and delete unmanaged config

Module will now return a list of unmanaged config. A delete_unmanaged
can also be set so that unmanaged config is deleted.

Also added a return for invalid commands.

samdoran commented:
This seems like a really risky behavior to have in a network module. It's good that the deletion is off by default, but still seems risky to have it. I don't know if there are other network modules with a similar behavior.

Commit:
updated match none test to use sort

On reviewing this test and the corresponding functionality it does not
appear to be used. The test itself also includes invalid config which
when fixed causes the test to fail.

Will raise an issue.

graemedavidson commented:
> This seems like a really risky behavior to have in a network module. It's good that the deletion is off by default, but still seems risky to have it. I don't know if there are other network modules with a similar behavior.

Hi @samdoran, 

I understand your thinking, I can remove the delete functionality and just retain the unmanaged warning if that is the preferred approach. Our team would prefer the delete as it stops rogue entries persisting but there are other ways of tackling that.

On another note reviewing my last commit and the following:
- https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/network/edgeos/edgeos_config.py#L283
- https://github.com/ansible/ansible/blob/devel/test/units/modules/network/edgeos/test_edgeos_config.py#L90

It appears to me the match param is not actually providing anything and the test currently fails due to not sorting. The config passed to it also appears invalid. Could you provide me any context and explanation for this just in case I am misunderstanding something.

samdoran commented:
You are correct: the `match` parameter is not being used. From looking at the `vyos_config` module, the `match` parameter is passed to the `get_diff()` method of the connection object. That method is missing from the EdgeOS cliconf plugin. It seems a lot has changed in the way network modules work since the EdgeOS modules were created and probably need some other updates as a result.

Commit:
added sort to returned list to ansible core

Improves readability

Commit:
removed now deprecated filtered tests

https://github.com/ansible/ansible/pull/63362

graemedavidson commented:
Have closed this for the time being and opened up another [PR](https://github.com/ansible/ansible/pull/67777) which just outputs unmanaged config in the module results.

We have used the output and ansible tasks to run deletes.
